pkg/envoy/envoy.go

```diff
@@ -174,14 +177,14 @@ func StartEnvoy(adminPort uint32, stateDir, logDir string, baseID uint64) *Envoy
 			default:
 			}
 
-			log.Info("Envoy: Process started at pid ", e.cmd.Process.Pid)
+			log.Info("Envoy: Process started at pid ", cmd.Process.Pid)
 
 			// We do not return after a successful start, but watch the Envoy process
 			// and restart it if it crashes.
-			err = e.cmd.Wait()
-			if err != nil {
+			if err := cmd.Wait(); err != nil {
```

Comment:
Unfortunately, doing this here will mean the goroutine will never see the stop signal (because it will say in the Wait until envoy stops. This only can happen if it crashes now). The arrangement here is still almost the same:
- make the wait a goroutine:
```golang
go func() {
if err := cmd.Wait(); err != nil {
log.WithError(err).Warn("Envoy: Execution failed.")
crashCh <- true // this can be whatever, it's just there to indicate to the select that envoy exited
e.errCh <- err
}
```

The error here is relevant since it's the error from envoy crashing. I'm not sure how to combine it with an error from the `Kill` in the select

- In the select add a case
```golang
case <-crashCh:
  // start envoy again
  continue // this isn't strictly needed, but it makes it clear we want the loop to happen again
```

Reply:
I don't think we need to send this error to the channel. Currently, we don't really handle the `err` object which is returned by `cmd.Wait()`, but instead we just log the fact that our process crashed.

But I agree with everything else, good point, I will submit a fix soon.